# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieJarSettings.h
# Commit: 92f3e33493ce
# Full Hash: 92f3e33493ce465875f6f959675119bdeddf3879
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2023-03-10 16:47:25
# Description:
#   Bug 1821545 - Reset the mFingerprintingRandomKey of CookieJarSettings before setting it. r=anti-tracking-reviewers,pbz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D172213
# ==============================================================================

diff -r f51c9e9c7a25 -r 92f3e33493ce netwerk/cookie/CookieJarSettings.h
--- a/netwerk/cookie/CookieJarSettings.h	Fri Mar 10 12:28:11 2023 +0000
+++ b/netwerk/cookie/CookieJarSettings.h	Fri Mar 10 12:29:21 2023 +0000
@@ -168,6 +168,8 @@
   const nsAString& GetPartitionKey() { return mPartitionKey; };
 
   void SetFingerprintingRandomizationKey(const nsTArray<uint8_t>& aKey) {
+    mFingerprintingRandomKey.reset();
+
     mFingerprintingRandomKey.emplace(aKey.Clone());
   }
 
